subsys/bluetooth/controller/multithreading_lock.c

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright (c) 2019 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+
+#include "multithreading_lock.h"
+
+/* Semaphore with a single count functioning as a lock. */
+K_SEM_DEFINE(ble_controller_lock, 1, 1);
+
+int32_t multithreading_lock_acquire(int32_t timeout)
+{
+#if IS_ENABLED(CONFIG_BLECTLR_THREADSAFE_BLOCKING)
+	return k_sem_take(&ble_controller_lock, K_MSEC(timeout));
+#else
+	return 0;
+#endif
+}
+
+int32_t multithreading_lock_acquire_try(void)
```

Comment:
Do we really need this API?

Reply:
We can remove this for now and add it later if required.